In [1]:
%load_ext autoreload
%autoreload 2

In [1]:
import pandas as pd
import json

import re
def load_json_field(bad_json):
    regex = r"\w\'\w"
    subst = ""
    bad_json = re.sub(regex, subst, bad_json)
    bad_json = bad_json.replace("d' Arrouzat", "darrouzat")
    good_json = bad_json.replace("\'", "\"").replace("None", "null").replace("True", "true").replace("False", "false")
    
    return json.loads(good_json)

data = pd.read_csv("bpideep/rawdata/data.csv")
data = data[data.deep_or_not != "almost_deeptech"]

In [2]:
data["team"] = data["team"].apply(load_json_field)
data["industries"] = data["industries"].apply(load_json_field)
data["investors"] = data["investors"].apply(load_json_field)

In [3]:
data.shape

(1339, 94)

Pour débugger :

nb = 0
for i in range(new.shape[0]):
    try :
        load_json_field(new["investors"][i])
    except:
        nb +=1
print(nb)

# Preprocessed data

Le model existant tel quel donne un X gardant la colonne "deep_or_not" -> enlève

In [5]:
data.team[0]

{'items': [], 'total': 0}

In [4]:
X = data.drop(columns = ['target'])
y = data['target']

# Train_test_split

In [5]:
from sklearn.model_selection import train_test_split

#train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, stratify = y)

#validation split
X_train_train, X_val, y_train_train, y_val = train_test_split(X_train, y_train, test_size=0.2, stratify = y_train)

# Existing model

In [6]:
from bpideep.trainer import Trainer

model = Trainer(X_train_train, y_train_train)
model.train()
model.pipeline.score(X_val, y_val)

set_pipeline 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degr

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))



degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['funding_employees_ratio'] = funding_amounts_employees(data)
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:263: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['has_strong_founder'] = data['has_strong_founder'].map({True : 1,
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:265: SettingWithCopyWarning: 
A value is trying to be 

growth_stage_num 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
indu

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 3.36
return_list 0.0
encoder 6.15
return_list 0.0
encoder 5.57
return_list 0.01
encoder 5.89
return_list 0.02
encoder 7.09
return_list 0.0
encoder 75.66
return_list 0.0
encoder 2.8
feat_eng_cols 107.47
train 107.51
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy 

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

0.6914893617021277

1st run : 0.6291079812206573

2nd run : 0.7276995305164319

In [7]:
from sklearn.model_selection import cross_validate

model2 = Trainer(X_train, y_train)
model2.train()

cv_result = cross_validate(model2.pipeline, X_train, y_train, cv=5, scoring=["accuracy", "recall", "f1", "precision"])

set_pipeline 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degr

degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
de

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))


degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['funding_employees_ratio'] = funding_amounts_employees(data)
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:263: SettingWithCopyWarning: 
A value is trying to be se


industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industrie

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:268: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['industry'] = data['industries'].map(lambda x: industries(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:270: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['year'] = pd.DataFrame({'year': data['launch_year']})
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:271: SettingWithCopyWarning: 
A value is trying to be set on a co

 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ra

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 4.09
return_list 0.0
encoder 8.96
return_list 0.0
encoder 7.64
return_list 0.01
encoder 9.74
return_list 0.02
encoder 10.89
return_list 0.01
encoder 133.98
return_list 0.0
encoder 3.12
feat_eng_cols 179.67
train 179.69
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
deg

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))


degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0

degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
funding_amounts_employees 0.0
growth_stage_num 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
in

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['funding_employees_ratio'] = funding_amounts_employees(data)
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:263: SettingWithCopyWarning: 
A value is trying to be se

 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ra

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 2.78
return_list 0.0
encoder 5.96
return_list 0.0
encoder 7.0
return_list 0.01
encoder 8.65
return_list 0.02
encoder 8.87
return_list 0.0
encoder 69.72
return_list 0.0
encoder 3.07
feat_eng_cols 106.87
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy 

 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ra

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))


0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_qua

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['funding_employees_ratio'] = funding_amounts_employees(data)
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:263: SettingWithCopyWarning: 
A value is trying to be se

 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
indus

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 4.8
return_list 0.0
encoder 10.8
return_list 0.0
encoder 6.35
return_list 0.01
encoder 5.86
return_list 0.02
encoder 7.2
return_list 0.01
encoder 80.31
return_list 0.0
encoder 2.91
feat_eng_cols 119.01
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy 

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_list 0.0
encoder 0.88
return_list 0.0
encoder 1.43
return_list 0.0
encoder 0.98
return_list 0.0
encoder 1.36
return_list 0.0
encoder 1.79
return_list 0.0
encoder 7.39
return_list 0.0
encoder 0.64
feat_eng_cols 14.63
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))


 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_qu

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['funding_employees_ratio'] = funding_amounts_employees(data)
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:263: SettingWithCopyWarning: 
A value is trying to be se


industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industrie

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 2.98
return_list 0.0
encoder 5.4
return_list 0.0
encoder 4.83
return_list 0.01
encoder 5.09
return_list 0.02
encoder 7.65
return_list 0.0
encoder 78.81
return_list 0.0
encoder 2.99
feat_eng_cols 108.54
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy 


degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_list 0.0


/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 1.2
return_list 0.0
encoder 1.25
return_list 0.0
encoder 1.1
return_list 0.0
encoder 1.4
return_list 0.0
encoder 2.08
return_list 0.0
encoder 10.1
return_list 0.0
encoder 1.58
feat_eng_cols 19.16
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degr

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy 

degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0

industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:268: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['industry'] = data['industries'].map(lambda x: industries(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:270: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['year'] = pd.DataFrame({'year': data['launch_year']})
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:271: SettingWithCopyWarning: 
A value is trying to be set on a co

0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_rat

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 3.93
return_list 0.0
encoder 5.85
return_list 0.0
encoder 8.13
return_list 0.01
encoder 6.06
return_list 0.02
encoder 7.86
return_list 0.01
encoder 81.1
return_list 0.0
encoder 2.82
feat_eng_cols 116.69
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy 

 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_list 0.0
encoder 0.75
return_list 0.0
encoder 1.14
return_list 0.0
encoder 1.15
return_list 0.0
encoder 1.34
return_list 0.0
encoder 1.9
return_list 0.0
encoder 5.99
return_list 0.0
encoder 0.66
feat_eng_cols 13.12
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy 

 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_qu

industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:268: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['industry'] = data['industries'].map(lambda x: industries(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:270: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['year'] = pd.DataFrame({'year': data['launch_year']})
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:271: SettingWithCopyWarning: 
A value is trying to be set on a co

return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

encoder 3.28
return_list 0.0
encoder 4.97
return_list 0.0
encoder 4.61
return_list 0.01
encoder 5.09
return_list 0.02
encoder 7.11
return_list 0.0
encoder 77.94
return_list 0.0
encoder 2.99
feat_eng_cols 106.85


/Users/catherinechen/.pyenv/versions/3.8.6/envs/deep4deep2/lib/python3.8/site-packages/sklearn/svm/_base.py:976: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn("Liblinear failed to converge, increase "
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/i

degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0

encoder 0.86
return_list 0.0
encoder 1.22
return_list 0.0
encoder 1.2
return_list 0.0
encoder 1.44
return_list 0.0
encoder 1.88
return_list 0.0
encoder 7.82
return_list 0.0
encoder 0.86
feat_eng_cols 15.49


In [8]:
scoring=["test_accuracy", "test_recall", "test_f1", "test_precision"]

for metric in scoring:
    print(f"{metric} : {cv_result[metric].mean() * 100 : .2f}%")

test_accuracy :  68.09%
test_recall :  57.84%
test_f1 :  63.67%
test_precision :  70.96%


1st run :
test_accuracy :  69.87%
test_recall :  49.22%
test_f1 :  58.09%
test_precision :  71.02%

2nd run (en instanciant bien model2) : 
test_accuracy :  69.96%
test_recall :  50.09%
test_f1 :  58.60%
test_precision :  70.77%

3rd run (drop almost deeptech):
test_accuracy :  68.09%
test_recall :  57.84%
test_f1 :  63.67%
test_precision :  70.96%

In [12]:
from sklearn.inspection import permutation_importance

permutation_score = permutation_importance(model2.train(), X_train, y_train, n_repeats=100)

np.vstack((X.columns, permutation_score.importances_mean)).T

set_pipeline 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degree 0.0
degr

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:260: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['degree'] = data['team'].map(lambda x:degree(x))


0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_qua

degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0.0
degree_quant 0

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:261: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['doctor_yesno'] = data['degree'].map(lambda x: degree_quant(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:262: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['funding_employees_ratio'] = funding_amounts_employees(data)
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:263: SettingWithCopyWarning: 
A value is trying to be se

 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
industries 0.0
indus

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:268: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['industry'] = data['industries'].map(lambda x: industries(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:270: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['year'] = pd.DataFrame({'year': data['launch_year']})
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:271: SettingWithCopyWarning: 
A value is trying to be set on a co

 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ratio 0.0
return_ra

/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:272: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['stage_age_ratio'] = data[['year_of_existence','growth_stage_num']]\
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:274: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data['investors_name'] = data['investors'].map(lambda x:investors_name(x))
/Users/catherinechen/code/bpifrance_deeptech_analysis/bpideep/feateng.py:275: SettingWithCopyWarning: 
A value is t

return_list 0.0
encoder 8.41
return_list 0.0
encoder 10.4
return_list 0.0
encoder 8.92
return_list 0.01
encoder 7.76
return_list 0.05
encoder 12.4
return_list 0.01
encoder 158.24
return_list 0.0
encoder 5.15
feat_eng_cols 212.62
train 212.65


TypeError: estimator should be an estimator implementing 'fit' method, None was passed